src/Compiler/Microsoft.AspNetCore.Razor.Language/src/DocumentationDescriptor.SimpleDescriptor.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using Microsoft.Extensions.Internal;
+
+namespace Microsoft.AspNetCore.Razor.Language;
+
+internal abstract partial class DocumentationDescriptor
+{
+    private sealed class SimpleDescriptor : DocumentationDescriptor
+    {
+        public override object?[] Args => Array.Empty<object>();
+
+        public SimpleDescriptor(DocumentationId id)
+            : base(id)
+        {
+        }
+
+        public override string GetText()
+            => GetDocumentationText();
+
+        public override bool Equals(DocumentationDescriptor other)
+            => other is SimpleDescriptor { Id: var id } && Id == id;
+
+        protected override int ComputeHashCode()
+        {
+            var result = HashCodeCombiner.Start();
+
+            result.Add((int)Id);
+
+            return result.CombinedHash;
+        }
```

Comment:
I'm curious why not just `Id.GetHashCode()`?

Reply:
A long while ago (.NET Framework 4.5?), the runtime would box an enum value when `GetHashCode()` was called. I'm pretty sure that was fixed long ago on NetFx, but I'm still a little allergic to the problem. Casting to an `int` is a completely reasonable hash code value for an enum, so I usually just do that. Happy to change it if you feel I'm a little stuck in the past. :smile:

Reply:
That makes sense, still it could be shortened to just `return ((int)Id).GetHashCode()` instead of using `HashCodeCombiner`, right? No changes needed, just wondering...

Reply:
Yes, good point. I'll do that since I'm still making tweaks.

Reply:
Note: Here's where my paranoia comes from: https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/Enum.cs,1254

Reply:
Looks like it was fixed on NetFx: https://referencesource.microsoft.com/#mscorlib/system/enum.cs,753